### PR TITLE
feat: add job live tracking

### PR DESCRIPTION
**Pull request by awaisdar001**
Add the endpoint for getting current geocode of the job assigned to a driver.

### CODE REVIEW

File: cigo_wrapper/network/CigoConnect.py [hunk 1]
```diff
@@ -179,5 +182,35 @@ def retrieve_a_itinerary(self, itinerary_id):
...
+            return self.__raise_response_exception(response)
+
+        job_data = data["job"]
+        job = Job.from_json(job_data)
+
+        if job_data.get("progress"):
+            job.progress = JobProgress.from_json(job_data["progress"])
+
+        if job_data.get("coordinates"):
+            job.coordinates = job_data["coordinates"]
```
Comment (AbdullahAlh) [inline cigo_wrapper/network/CigoConnect.py:206]: This is not needed as it is assigned in `job = Job.from_json(job_data)`

File: cigo_wrapper/network/CigoConnect.py [hunk 2]
```diff
@@ -179,5 +182,35 @@ def retrieve_a_itinerary(self, itinerary_id):
...
+        data = response.json()
+
+        if data['statusCode'] != 200:
+            return self.__raise_response_exception(response)
+
+        job_data = data["job"]
+        job = Job.from_json(job_data)
+
+        if job_data.get("progress"):
+            job.progress = JobProgress.from_json(job_data["progress"])
```
Comment (AbdullahAlh) [inline cigo_wrapper/network/CigoConnect.py:203]: This is not needed as it is assigned in job = Job.from_json(job_data)

File: cigo_wrapper/entity/JobGeocoding.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    duration = None
+    start_datetime = None
+    end_datetime = None
+    started = None
+    finished = None
+
+    def __init__(self, validity="", subdivision_code="", country_code=""):
+        self.validity = validity
+        self.subdivision_code = subdivision_code
+        self.country_code = country_code
```
[the remark this answers is not in the document]
Comment (awaisdar001) [reply cigo_wrapper/entity/JobGeocoding.py]: @AbdullahAlh do you mean `country_code` or `JobGeocoding` class?

File: cigo_wrapper/network/CigoConnect.py [hunk 4]
```diff
@@ -179,5 +182,35 @@ def retrieve_a_itinerary(self, itinerary_id):
...
+
+        if job_data.get("coordinates"):
+            job.coordinates = job_data["coordinates"]
+
+        if job_data.get("geocoding"):
+            job.geocoding = JobGeocoding.from_json(job_data["geocoding"])
```
[the remark this answers is not in the document]
Comment (awaisdar001) [reply cigo_wrapper/network/CigoConnect.py:209]: This is same as `coordinates` and `geocoding`.

File: cigo_wrapper/network/CigoConnect.py (hunk 2 above)
Comment (awaisdar001) [reply cigo_wrapper/network/CigoConnect.py:203]: This is same as `coordinates`,  `geocoding` and `progress`.

**Comment by awaisdar001** [comment]
@AbdullahAlh I have responded on the PR. Can you please check the PR again when its possible? thanks.

File: cigo_wrapper/network/CigoConnect.py (hunk 1 above)
Comment (AbdullahAlh) [reply cigo_wrapper/network/CigoConnect.py:206]: @awaisdar001 can we add a comment explaining this pls

**Comment by awaisdar001** [comment]
@AbdullahAlh changes are done, please take another look.

File: cigo_wrapper/network/CigoConnect.py (hunk 1 above)
Comment (awaisdar001) [reply cigo_wrapper/network/CigoConnect.py:206]: Yes I can add that comment 👍

**Review by AbdullahAlh — APPROVED** [review]

**Pull request merged**